RELEASE.md

```diff
@@ -2,6 +2,43 @@
 
 This document guides a contributor through creating a release of JupyterLab.
 
+## The JupyterLab Release Process
+
+JupyterLab follows [semver](https://semver.org/) for the versioning of the Python and JavaScript packages.
+
+### Release Timeline
+
+Although the commitments listed below are "best effort", the JupyterLab team tries to follow a couple of guidelines:
+
+- one major version per year, which usually includes API breaking changes
+- several minor versions per year that include new features but no API breaking changes
+- support and bug fixes on a couple of final releases (by backporting PRs and releasing from release branches)
+
+Release Plans are tracked in dedicated issues, and are closed when the final release. See the following two issues as an example:
+
+- [3.3 Release Plan](https://github.com/jupyterlab/jupyterlab/issues/11643)
+- [4.0 Release Plan](https://github.com/jupyterlab/jupyterlab/issues/9647)
+
+### Alpha Releases
+
+Alpha releases have a fairly low bar. Their purpose is to start putting the new JupyterLab version into the hands of users and extension authors.
+
+The requirements for an alpha release should be that JupyterLab can be installed and run. Bugs and breaking changes are accepted.
+
+### Beta Releases
+
+Beta releases usually try to not have breaking changes in the API, although breaking changes can sometimes happen during that phase if they were missed during the alpha stage.
+
+The recommended time period for the Beta phase is a minimum of 1 week.
+
+### Release Candidates
+
+Release Candidates (RC) are a signal to the extension developer community that they should start migrating to the new version to test it. At that point we consider the software stable.
+
+The RC stage is often a good time to address final release documentation changes or minor UX tweaks. During the RC phase, the JupyterLab developers and maintainers start updating third-party extensions over to the new version to test it. This work during the RC phase, and giving time for feedback from extension developers, can take up to a couple of weeks.
+
+The recommended time period for the Release Candidate phase is a minimum of 1 week for minor releases, and 2 weeks for major releases.
+
 ## Automated Releases with the Jupyter Releaser
```

Comment:
The docs already include youtube videos. Maybe we could add this one https://www.youtube.com/watch?v=cdRvvyZvYKM&t=333s if @blink1073 plans to leave it online for some time;

Reply:
The YoutTube walk-through is already mentioned two lines below.